scrape_books/pipelines.py

```diff
@@ -0,0 +1,30 @@
+from scrapy import Spider
+from typing import Any, Dict
+
+
+RATING: Dict[str, int] = {
+    "One": 1,
+    "Two": 2,
+    "Three": 3,
+    "Four": 4,
+    "Five": 5
+}
+
+
+class ScrapeBooksPipeline:
+    def process_item(
+            self, item: Dict[str, Any], spider: Spider
+    ) -> Dict[str, Any]:
+        item["price"] = float(item["price"].replace("£", ""))
+
+        if len(item["amount_in_stock"]) > 0:
+            item["amount_in_stock"] = int(
+                item["amount_in_stock"][1].replace("\n", "")
+                .split()[2].replace("(", "")
+            )
+        else:
+            item["amount_in_stock"] = 0
+
+        item["rating"] = RATING[item["rating"].split()[1]]
```

Comment:
The `rating` extraction assumes that the rating string always contains a word that matches the keys in the `RATING` dictionary. Ensure that this assumption holds true for all input data to prevent `KeyError` exceptions.